rust/agama-server/src/agama-web-server.rs

```diff
@@ -39,26 +40,270 @@ struct Cli {
     pub command: Commands,
 }
 
+#[derive(Args, Debug)]
+struct ServeArgs {
+    // Address/port to listen on (":::3000" listens for both IPv6 and IPv4
+    // connections unless manually disabled in /proc/sys/net/ipv6/bindv6only)
+    #[arg(long, default_value = ":::3000", help = "Primary address to listen on")]
+    address: String,
+    #[arg(
+        long,
+        default_value = "",
```

Comment:
Please, do not use "empty" values to mean "nothing". Prefer using an `Option<T>`.